src/api/app/models/project/remote_url.rb

```diff
@@ -2,11 +2,21 @@ class Project::RemoteURL
   require 'open-uri'
 
   def self.load(remote_project, path)
+    if File.file?('/etc/sysconfig/proxy')
+      proxysettings = Hash[File.read('/etc/sysconfig/proxy').scan(/(\S+)\s*=\s*"([^"]+)/)]
+      Rails.logger.info "var HTTP_PROXY: #{proxysettings.fetch("HTTP_PROXY")} var HTTPS_PROXY: #{proxysettings.fetch("HTTPS_PROXY")} var NO_PROXY: #{proxysettings.fetch("NO_PROXY")}"
```

Comment:
Using `fetch` will raise a `KeyError` exception if a key isn't present. This should be changed:

```suggestion
      Rails.logger.info "var HTTP_PROXY: #{proxysettings['HTTP_PROXY']} var HTTPS_PROXY: #{proxysettings['HTTPS_PROXY']} var NO_PROXY: #{proxysettings['NO_PROXY']}"
```

I don't know if this is intended, but `FTP_PROXY` is missing in that line.

Reply:
The code has been fixed and tested.